### ISSUE TITLE
Parser error with brackets

### ISSUE THREAD

**Issue by patrick-kidger** [google-research/dex-lang#997]
```
data LinearStruct = MkLinearStruct (input:Type) (output:Type) (weight:output=>input=>Float) (bias:output=>Float)

> Parse error:3:47:
>   |
> 3 | data LinearStruct = MkLinearStruct (input:Type) (output:Type) (weight:output=>input=>Float) (bias:output=>Float)
>   |                                               ^^^
> unexpected ") ("
> expecting "=>" or arrow
```

Removing the brackets around `(input:Type)` and `(output:Type)` might fix things?

**Comment by dougalm** [comment]
Yes, for now you need write it without the brackets. The treatment of brackets is a complete mess at the moment, with contradictory conventions in different parts of the syntax, see #953. @axch is working on a PR to fix this, #965.

**Comment by patrick-kidger** [comment]
Haha yup -- @axch asked me to open the issue to track this ;)
